File: backend/app/contexts/teams/interface/http/v1/schemas/teams.py
"""Team lifecycle and summary HTTP schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TeamSummary(BaseModel):
    id: int
    name: str
    description: str | None = None
    hospital: str | None = None
    department: str | None = None
    creator_name: str | None = None
    member_count: int = 0
    max_members: int | None = None
    is_member: bool = False
    my_role: str | None = None
    my_status: str | None = None
    is_creator: bool = False
    join_status: str | None = None
    join_request_id: int | None = None
    created_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class TeamCreateRequest(BaseModel):
    name: str = Field(..., min_length=2, max_length=120)
    description: str | None = Field(None, max_length=1000)
    hospital: str | None = Field(None, max_length=120)
    department: str | None = Field(None, max_length=120)
    max_members: int | None = Field(50, ge=1, le=500)


class TeamUpdateRequest(BaseModel):
    name: str | None = Field(None, min_length=2, max_length=120)
    description: str | None = Field(None, max_length=1000)
    hospital: str | None = Field(None, max_length=120)
    department: str | None = Field(None, max_length=120)
    max_members: int | None = Field(None, ge=1, le=500)

File: packages/xiehe-api-contracts/src/http/v1/teams.ts
export type TeamRole = 'ADMIN' | 'MEMBER';

export interface TeamSummary {
  id: number;
  name: string;
  description?: string | null;
  hospital?: string | null;
  department?: string | null;
  creator_name?: string | null;
  member_count: number;
  max_members?: number | null;
  is_member: boolean;
  my_role?: TeamRole | null;
  my_status?: 'ACTIVE' | 'INVITED' | 'PENDING' | 'INACTIVE' | null;
  is_creator?: boolean;
  join_status?: string | null;
  join_request_id?: number | null;
  created_at?: string | null;
}

export interface TeamListResponse {
  items: TeamSummary[];
  total: number;
}

export interface TeamSearchResponse {
  results: TeamSummary[];
  total: number;
}

export interface TeamMember {
  user_id: number;
  username: string;
  real_name?: string | null;
  email?: string | null;
  role: TeamRole;
  status: string;
  department?: string | null;
  is_creator: boolean;
  is_system_admin: boolean;
  system_admin_level: number;
  joined_at?: string | null;
}

export interface TeamMembersResponse {
  team: TeamSummary;
  members: TeamMember[];
}

export interface TeamJoinRequestItem {
  id: number;
  team_id: number;
  applicant_id: number;
  applicant_username: string;
  applicant_real_name?: string | null;
  applicant_email?: string | null;
  message: string;
  status: 'PENDING' | 'APPROVED' | 'REJECTED' | 'CANCELLED';
  requested_at: string;
  reviewed_at?: string | null;
  reviewer_id?: number | null;
}

export interface TeamJoinRequestListResponse {
  items: TeamJoinRequestItem[];
  total: number;
  pending_count: number;
}

export interface TeamJoinRequestActionResponse {
  message: string;
  status: 'approved' | 'rejected' | 'pending';
  request: TeamJoinRequestItem;
}

export interface TeamJoinRequestSubmitResponse {
  request_id: number;
  message: string;
  status: string;
  requested_at: string;
}

export interface TeamCreateRequest {
  name: string;
  description?: string;
  hospital?: string;
  department?: string;
  max_members?: number;
}

export interface TeamUpdateRequest extends Partial<TeamCreateRequest> {}

export interface TeamInvitationItem {
  id: number;
  team_id: number;
  team_name?: string | null;
  team_description?: string | null;
  inviter_id: number;
  inviter_name?: string | null;
  role: string;
  message?: string | null;
  created_at: string;
  expires_at: string;
  status: string;
}

export interface TeamInvitationListResponse {
  items: TeamInvitationItem[];
  total: number;
}

export interface TeamInvitationRespondResponse {
  message: string;
  status: string;
  team_id: number;
  team_name: string;
}
